contracts/BSCValidatorSet.sol

```diff
@@ -208,6 +228,15 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
       return ERROR_FAIL_CHECK_VALIDATORS;
     }
 
+    // step 0: force all maintaining validators to exit `Temporary Maintenance`
+    address validator;
+    for (uint i = 0; i < maintainingValidatorSet.length; i++) {
+      validator = maintainingValidatorSet[i].consensusAddress;
+      _exitMaintenance(validator, payable(BURN_ADDRESS));
+      delete maintainInfoMap[validator];
+    }
+    delete maintainingValidatorSet;
```

Comment:
Are you sure we can simply delete an slice in this way?

Reply:
yes